packages/flutter_tools/lib/src/ios/devices.dart

```diff
@@ -548,14 +560,14 @@ class _IOSDeviceLogReader extends DeviceLogReader {
 
   void _start() {
     iMobileDevice.startLogger(device.id).then<void>((Process process) {
-      _process = process;
-      _process.stdout.transform<String>(utf8.decoder).transform<String>(const LineSplitter()).listen(_newLineHandler());
-      _process.stderr.transform<String>(utf8.decoder).transform<String>(const LineSplitter()).listen(_newLineHandler());
-      _process.exitCode.whenComplete(() {
+      process.stdout.transform<String>(utf8.decoder).transform<String>(const LineSplitter()).listen(_newLineHandler());
```

Comment:
Since these pids are created by the log reader/portfowarder implementations, they should be disposed by those implementations as well - adding `killSubProcesses` feels a bit leaky. I would suggest a slightly different arrangement:

1. Add a `dispose` method to the LogReader and PortForwarder interfaces.
2. Either cache the reader/port forwarder in the device, and likewise add a dispose to device, or otherwise hold onto these objects to dispose them.

Reply:
done